### ISSUE TITLE
sparsify(MX): identity

### ISSUE THREAD

**Issue by jgillis** [casadi/casadi#1381]
For the sake of symmetry between SX/MX, we need to have sparsify not trhow an error, but be an identity transformation

**Comment by jaeandersson** [comment]
:+1:

**Comment by ghorn** [comment]
I believe there are some cases when you could do more than identity, such as `sparsify(f(MX::sym()))` where `f` is a `Function` with sparse outputs

**Comment by jaeandersson** [comment]
> I believe there are some cases when you could do more than identity...

Obviously, there are some cases where more sparsity is available. The `SetSparse` node is used to make an expression more dense than otherwise needed.

**Comment by jaeandersson** [comment]
You can do a sparsify(MX) implementation that will work just like sparsify(SX). What you need to do is to create an MXFunction and evaluate it using "NaN" for all inputs. If something depends on an input, it will be NaN. But if something just depends on constants, it will be a number which can be compared with 0.

**Comment by jgillis** [comment]
Sounds hacky. Any second thoughts before i give this a shot?

**Comment by jaeandersson** [comment]
Yeah, don't implement that. Have the identity transformation if anything. No hack.

**Comment by jaeandersson** [comment]
Or rather, the correct way of implement this with a virtual function `getSparsify` in the MXNode class. This can be overloaded, e.g. for constants for "setSparse` nodes.
